configuration/parser.go

```diff
@@ -206,8 +206,10 @@ func (p *Parser) overwriteStruct(v reflect.Value, fullpath string, path []string
 		fieldVal := reflect.New(sf.Type)
 		err := yaml.Unmarshal([]byte(payload), fieldVal.Interface())
 		if err != nil {
-			logrus.Warnf("Error parsing environment variable %s: %s", fullpath, err)
-			return err
+			return &ErrCannotParseEnvVar{
```

Comment:
This is a bit of a departure, but could we instead use the error here with more generic text here: https://github.com/distribution/distribution/pull/3680/files#diff-11c09f5dd24470c6ed1a379ca9301e7b60a48918abc110c57d5c4bd303f88bbfR141

That should catch every error that comes with overriding the config with ENVARs

Reply:
I wasn't 100% sure what you meant here, but I think its this.